alf/metrics/metrics.py

```diff
@@ -145,33 +146,35 @@ def __init__(self,
                  name="AverageEpisodicAggregationMetric",
                  prefix='Metrics',
                  dtype=torch.float32,
-                 batch_size=1,
                  buffer_size=10,
-                 example_metric_value=None):
+                 example_time_step=None):
         """
         Args:
             name (str):
             prefix (str): a prefix indicating the category of the metric
             dtype (torch.dtype): dtype of metric values. Should be floating types
                 in order to be averaged.
-            batch_size (int): the number of metric values generated in parallel
             buffer_size (int): number of episodes the metric value will be averaged
                 across
-            example_metric_value (nest): an example of metric value to be summarized;
-                if ``None``, a zero scalar is used.
+            example_time_step (nest): an example of the time step where the metric
+               values are extracted from. If ``None``, a zero scalar is used as the
+               example metric value.
         """
         super(AverageEpisodicAggregationMetric, self).__init__(
             name=name, dtype=dtype, prefix=prefix)
-        if example_metric_value is None:
+        if example_time_step is None:
             example_metric_value = torch.zeros((), device='cpu')
-        self._batch_size = batch_size
+        else:
+            example_metric_value = self._extract_metric_values(
+                alf.nest.utils.convert_device(example_time_step, device='cpu'))
```

Comment:
Not sure why the original example_metric_value is created on cpu. It's not necessary as I see. Only its structure (not its values) is used.

Reply:
I think it is because of `AverageDicountedReturnMetric`'s `accumulated_discount`.

Reply:
Can directly use `example_time_step.cpu()`

Reply:
Fixed.